### ISSUE TITLE
MeCab plugin: support extracting specific part-of-speech

### ISSUE THREAD

**Issue by kmaehashi** [jubatus/jubatus#1073]
Using specific type of part-of-speech (`品詞`) to construct feature vector is a common requirement.
For example, users may want to only use nouns to extract topics from the document.

Currently we users need to do that in client side, which is costful.  So I want to add that feature to MeCab plugin.

As words in MeCab dictionaries have hierarchical (4 layered) part-of-speech (e.g., `名詞/固有名詞/人名/姓`), we'll need to design some kind of simple query language in JSON, like this:

```
{
  "include": ["名詞/*", "動詞/*"],
  "exclude": ["名詞/固有名詞/*"]
}
```

**Comment by kmaehashi** [comment]
Refer to this page for the hierarchy of part-of-speeches:
http://www.unixuser.org/~euske/doc/postag/#chasen

**Comment by kmaehashi** [comment]
Currently plug-in parameters are only given as `map<string,string>`:

```
jubatus::plugin::fv_converter::mecab_splitter* create(
    const std::map<std::string, std::string>& params);
```

which does not allow passing lists.

Taking this limitation into consideration, I propose to add two new parameters to the mecab plug-in:

```
{
  "include_features": "名詞,*|動詞,*",
  "exclude_features": "名詞,固有名詞,*|名詞,接尾,*|動詞,非自立,*"
}
```

which behaves as follows:

1. Extract words with features that matches patterns in `include_features` (separated by `|`).  By default `include_features` is `*` (include everything).
2. From words extracted in 1., filter out words with features that matches patterns in `exclude_features` (separated by `|`).  By default `exclude_features` is `` (exclude nothing).

Note: in this discussion, the term `features` refers to a CSV-formatted mecab output (see the example below):

```
$ mecab
特徴を抽出した
特徴    名詞,一般,*,*,*,*,特徴,トクチョウ,トクチョー
を      助詞,格助詞,一般,*,*,*,を,ヲ,ヲ
抽出    名詞,サ変接続,*,*,*,*,抽出,チュウシュツ,チューシュツ
し      動詞,自立,*,*,サ変・スル,連用形,する,シ,シ
た      助動詞,*,*,*,特殊・タ,基本形,た,タ,タ
EOS
```

Note: patterns matching is done using existing key matcher as in `string_filter_rules`. For details see: http://jubat.us/ja/fv_convert.html#string-filter-rules

**Comment by kmaehashi** [comment]
Fixed via #1098
